src/enumerable.cr

```diff
@@ -644,6 +664,28 @@ module Enumerable(T)
     index { |e| e == obj }
   end
 
+  # Returns the index of the first element for which the passed block returns `true`.
+  #
+  # ```
+  # ["Alice", "Bob"].index! { |name| name.size < 4 } # => 1 (Bob's index)
+  # ```
+  #
+  # Raises `Enumerable::NotFoundError` if there are no element for which the block is `true`.
+  def index!(& : T ->) : Int32
+    index { |e| yield e } || raise Enumerable::NotFoundError.new
+  end
+
+  # Returns the index of the object *obj* in the collection.
```

Comment:
If multiple elements equal `obj`, only the first index is returned.

"the object": Unnecessary specification 
```suggestion
  # Returns the first index of *obj* in the collection.
```